Slider/Assets/Scripts/UI/CutsceneManager.cs

```diff
@@ -87,26 +80,79 @@ public void showImages()
     IEnumerator scrolltext(GameObject text)
     {
         skipImages = false;
-        TMP_Text tmp = text.GetComponent<TMP_Text>();
-
-        string s1 = tmp.text;
-        string s2 = "";
-        tmp.text = "";
+        TMPTextTyper tmp_typer = text.GetComponent<TMPTextTyper>();
+        string s1 = textboxes[i].GetComponent<TextMeshProUGUI>().text;
+        textboxes[i].GetComponent<TextMeshProUGUI>().text = "";
         //wait a bit so it doesnt start during fadein
         yield return new WaitForSeconds(0.2f);
-        foreach (char letter in s1)
+        tmp_typer.StartTyping(s1);
+        while (!tmp_typer.finishedTyping)
         {
             if (skipImages)
             {
-                tmp.text = s1;
-                break;
+                if (tmp_typer.TrySkipText()) break;
             }
-            s2 += letter;
-            tmp.text = (s2);
+            yield return null;
+        }
+        skipImages = false;
+        StartCoroutine(WaitToAdvance());
+    }
 
-            yield return new WaitForSeconds(2 * GameSettings.textSpeed);
+    IEnumerator BoxFadeIn()
+    {
+        Color color = blackBox.color;
```

Comment:
I think we should use a CanvasGroup object on all of the texts/images to handle fading, rather than a black panel. Then you can update the CanvasGroup.alpha to update it's transparency. (to do this you would make an empty gameobj, make Images and Texts childed to that, and at the CanvasGroup component to the new obj)

Alternatively, it might also work to just set the background box colors to the slider black instead of pure black